### PR TITLE
JAMES-3715 Add IMAP IDLE tests when combined with SSL, COMPRESS

### PR DESCRIPTION
**Pull request by chibenwa**
## Why

Screenshot from 2022-03-16 16-12-43

This reliable reproduces the issue in the Netty4 pull request.

### CODE REVIEW

File: server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java [hunk 1]
```diff
@@ -1026,6 +1041,241 @@ void searchingASingleUTF8CriterionShouldComplete() throws Exception {
...
+            imapServer = createImapServer("imapServerSSL.xml");
+            int port = imapServer.getListenAddresses().get(0).getPort();
+            mailboxSession = memoryIntegrationResources.getMailboxManager().createSystemSession(USER);
+            memoryIntegrationResources.getMailboxManager()
+                .createMailbox(MailboxPath.inbox(USER), mailboxSession);
+            inbox = memoryIntegrationResources.getMailboxManager().getMailbox(MailboxPath.inbox(USER), mailboxSession);
+
+            SSLContext ctx = SSLContext.getInstance("TLS");
+            ctx.init(null, new TrustManager[] { new BlindTrustManager() }, null);
+            clientConnection =ctx.getSocketFactory().createSocket();
```
Comment (quantranhong1999) [inline server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java:1077]: a little space here ^^

File: server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java [hunk 2]
```diff
@@ -1026,6 +1041,241 @@ void searchingASingleUTF8CriterionShouldComplete() throws Exception {
...
+            Awaitility.await().atMost(Duration.ofSeconds(2)).untilAsserted(() ->
+                assertThat(readStringUntil(clientConnection, s -> s.contains("* 1 EXISTS")))
+                    .isNotNull());
+        }
+
+        private byte[] readBytes(Socket channel) throws IOException {
+            byte[] buffer = new byte[8193];
+            final int read = channel.getInputStream().read(buffer);
+            final String s = new String(buffer, 0, read, StandardCharsets.US_ASCII);
+            System.out.println("C: " + s);
```
Comment (quantranhong1999) [inline server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java:1193]: Please remove this after finishing debug

File: server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java [hunk 3]
```diff
@@ -1026,6 +1041,241 @@ void searchingASingleUTF8CriterionShouldComplete() throws Exception {
...
+        void idleShouldSendInitialContinuation() throws Exception {
+            clientConnection.getOutputStream().write(String.format("a0 LOGIN %s %s\r\n", USER.asString(), USER_PASS).getBytes(StandardCharsets.UTF_8));
+            readBytes(clientConnection);
+
+            clientConnection.getOutputStream().write("a2 SELECT INBOX\r\n".getBytes(StandardCharsets.UTF_8));
+            readStringUntil(clientConnection, s -> s.contains("a2 OK [READ-WRITE] SELECT completed."));
+
+            clientConnection.getOutputStream().write("a3 IDLE\r\n".getBytes(StandardCharsets.UTF_8));
+
+
```
Comment (quantranhong1999) [inline server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java:1099]: We could reduce a line here?

**Comment by Arsnael** [comment]
```
org.apache.james.imapserver.netty.IMAPServerTest$AuthenticationRequireSSL.loginShouldFailWhenRequireSSLAndUnEncryptedChannel

java.lang.RuntimeException: Unable to register mbean
	at org.apache.james.imapserver.netty.IMAPServerTest$AuthenticationRequireSSL.loginShouldFailWhenRequireSSLAndUnEncryptedChannel(IMAPServerTest.java:775)
Caused by: javax.management.InstanceAlreadyExistsException: org.apache.james:type=server,name=imapserver,sub-type=threadpool,threadpool=imapserver-executor
	at org.apache.james.imapserver.netty.IMAPServerTest$AuthenticationRequireSSL.loginShouldFailWhenRequireSSLAndUnEncryptedChannel(IMAPServerTest.java:775)
```

**Review by Arsnael — COMMENTED** [review]
Read it nothing to add

File: server/protocols/protocols-imap4/pom.xml [hunk 4]
```diff
@@ -131,6 +131,16 @@
             <groupId>io.netty</groupId>
             <artifactId>netty</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-all</artifactId>
+            <version>4.1.72.Final</version>
```
Comment (vttranlina) [inline server/protocols/protocols-imap4/pom.xml:137]: should we have a property for all netty dependency?

Comment (chibenwa) [reply server/protocols/protocols-imap4/pom.xml:137]: This is a testing dependency and this will disapear with the netty4 migration...

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**